packages/@aws-cdk/aws-cloudfront/lib/key-group.ts

```diff
@@ -0,0 +1,73 @@
+import { Names, Resource, Token } from "@aws-cdk/core";
+import { Construct } from 'constructs';
+import { CfnKeyGroup } from './cloudfront.generated';
+
+/**
+ * Represents a Key Group
+ */
+export interface IKeyGroup {
+  /**
+   * The ID of the key group.
+   * @attribute
+   */
+  readonly keyGroupId: string;
+}
+
+/**
+ * Properties for creating a Key Group
+ */
+export interface KeyGroupProps {
+  /**
+   * A name to identify the key group.
+   * @default - generated from the `id`
+   */
+  readonly keyGroupName?: string;
+  
+  /**
+   * A comment to describe the key group.
+   * @default - no comment
+   */
+  readonly comment?: string;
+
+  /**
+   * A list of the identifiers of the public keys in the key group.
+   */
+  readonly items?: string[];
```

Comment:
Let's keep this typed (IPublicKey) and make it mandatory; I don't see any reason to create an empty KeyGroup. 

```suggestion
  readonly keys: IPublicKey[];
```

Reply:
Yep... I will address this. This was just a placeholder. This is where I've decided to lump `PublicKey` and `KeyGroup` implementations together rather than separate PRs.

Reply:
Switching this to a `IPublicKey` requires proper management of the parameter in the constructor (array of strings) which will probably be a helper method.